src/clj/game/core/runs.clj

```diff
@@ -44,7 +44,9 @@
   (when (and (is-type? c "Agenda")
              (not (find-cid (:cid c) (get-in @state [:runner :scored]))))
     (trigger-event state side :no-steal c))
-  (when (get-card state c)
+  (when (and (get-card state c)
+             ;; Don't increment :no-trash-or-steal if accessing a card in Archives
+             (not (find-cid (:cid c) (get-in @state [:corp :discard]))))
```

Comment:
Can this be done a little more directly, like checking the zone of `c`? Also, couldn't we predicate this on `c` not being an agenda?

Reply:
Yeah, checking the zone is better. Forgot about that.

We still want it increment when you see an agenda, as that's what Maw looks for. Otherwise, declining to steal an agenda with an additional cost won't trigger Maw.